### ISSUE TITLE
Independent data sampling between domains

### ISSUE THREAD

**Issue by razorx89** [icon-lab/SynDiff#4]
It seems like currently during training the same image pairs are generated, since only one dataset is defined.

https://github.com/icon-lab/SynDiff/blob/375af793b5e2de1dfac7046249edbdb19e58a7f5/dataset.py#L14

https://pytorch.org/docs/stable/data.html#torch.utils.data.TensorDataset

I think this should be implemented using two data loaders and two datasets, which then independently sample from each domain and always create unique pairs after shuffling. This could also improve the overall generalizability.
